### PR TITLE
Remove condition to disable transaction confirm button if user has no balance

### PR DESCRIPTION
**Pull request by jpuri**
## **Description**
Currently we have these 2 conditions checked to disable submit button on confirmation pages:

1. account balance is less that gas + transaction value or
2. account balance is `0`

The second condition disables a `0` value transaction on a gasless network - something we support in MetaMask extension.

The PR removes the second condition and the first one alone is sufficient.

## **Manual testing steps**

1. Follow the steps listed here https://consensys.net/docs/goquorum/en/latest/tutorials/quorum-dev-quickstart/ to run Quorum locally
2. Run a DAPP locally and send transaction with value and gasPrice to `0`
3. Notice the "Submit" button is enabled

## **Related issues**
_Fixes #6567

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained:
  - [x] What problem this PR is solving.
  - [x] How this problem was solved.
  - [x] How reviewers can test my changes.
- [x] I’ve indicated what issue this PR is linked to: Fixes #???
- [x] I’ve included tests if applicable.
- [ ] I’ve documented any added code.
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
- [x] I’ve properly set the pull request status:
  - [x] In case it's not yet "ready for review", I've set it to "draft".
  - [x] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: app/components/UI/TransactionReview/index.js [hunk 1]
```diff
@@ -514,10 +503,7 @@ class TransactionReview extends PureComponent {
               onConfirmPress={this.props.onConfirm}
               confirmed={transactionConfirmed}
               confirmDisabled={
-                senderBalanceIsZero ||
-                transactionConfirmed ||
-                Boolean(error) ||
-                isAnimating
+                transactionConfirmed || Boolean(error) || isAnimating
```
Comment (blackdevelopa) [inline app/components/UI/TransactionReview/index.js:506]: We are removing the check for the account balance when submitting a transaction. Looks good.

**Review by blackdevelopa — DISMISSED** [review]

**Review by digiwand — DISMISSED** [review]
Lgtm! can merge after `ci / scripts (audit:ci)` test works again. more details can be found [here](https://consensys.slack.com/archives/C02U025CVU4/p1695892260000329)

**Comment by jpuri** [comment]
Bitrise: https://app.bitrise.io/app/be69d4368ee7e86d/pipelines/29631341-e3ce-4196-9ec5-fff489888d54

**Review by blackdevelopa — DISMISSED** [review]

**Review by digiwand — COMMENTED** [review]
just curious what was the thought around removing `senderBalanceIsZero` entirely? Should we preserve this for networks we know that use gas?

**Review by segun — DISMISSED** [review]

**Review by segun — APPROVED** [review]

**Review by blackdevelopa — APPROVED** [review]

**Pull request merged**